docs/ai/tutorials/evaluate-with-reporting.md

```diff
@@ -0,0 +1,158 @@
+
+1. Add a method to validate the response.
+
+   :::code language="csharp" source="./snippets/evaluate-with-reporting/MyTests.cs" id="Validate":::
+
+    > [!TIP]
+    > The relevance, truth, and completeness metrics each include a `Reason` property that explains the reasoning for the score. The reason is included in the generated report and can be viewed in the raw JSON file or by hovering over the corresponding metric's card in the report.
+
+1. Finally, add the [test method](xref:Microsoft.VisualStudio.TestTools.UnitTesting.TestMethodAttribute) itself.
+
+   :::code language="csharp" source="./snippets/evaluate-with-reporting/MyTests.cs" id="TestMethod":::
+
+   This test method:
+
+   - Creates the <xref:Microsoft.Extensions.AI.Evaluation.Reporting.ScenarioRun>.
+   - Gets the LLM's response to a specific astronomy question. Since response caching is turned on for the reporting configuration, the LLM response will be fetched either:
+     - Directly from the LLM endpoint in the very first run of the current test.
+     - From the (disk-based) response cache that was configured in `s_defaultReportingConfiguration` in every subsequent run of the test, until the cached entry expires (in 14 days by default).
+   - Runs the evaluators against the response. Since response caching is enabled, the evaluation will be either:
+     - Performed using the the LLM endpoint in the very first run of the current test.
+     - Fetched from the (disk-based) response cache that was configured in `s_defaultReportingConfiguration` in every subsequent run of the test, until the cached entry expires (in 14 days by default).
+   - Runs some basic validation on the evaluation result.
+
+     This step is optional and mainly for demonstration purposes. In real-world evaluations, you might not want to validate individual results since the LLM responses and evaluation scores can change over time as your product (and the models used) evolve. You might not want individual evaluation tests to "fail" and block builds in your CI/CD pipelines when this happens. Instead, it might be better to rely on the generated report and track the overall trends for evaluation scores across different scenarios over time (and only fail individual builds when there's a significant drop in evaluation scores across multiple different tests). That said, there is some nuance here and the choice of whether to validate individual results or not can vary depending on the specific use case.
+
+   When the method returns, the `scenarioRun` object is disposed and the evaluation result for the evaluation is stored to the (disk-based) result store that's configured in `s_defaultReportingConfiguration`.
+
+## Run the test/evaluation
+
+Run the test using your preferred test workflow, for example, by using the CLI command `dotnet test` or through [Test Explorer](/visualstudio/test/run-unit-tests-with-test-explorer).
```

Comment:
It would be great to add another step that shows how to view the report by installing and running the `dotnet aieval` command line tool. We can also include a screenshot of what this reporting for the metrics looks like when you expand the scenario tree in the report to view the results of the above test.